commit b288052e1779261ae80138074989ef50358c4e58
Author: Chris Mason <chris.mason@oracle.com>
Date:   Thu Feb 12 09:37:35 2009 -0500

    Btrfs: process mount options on mount -o remount,
    
    Btrfs wasn't parsing any new mount options during remount, making it
    difficult to set mount options on a root drive.
    
    Signed-off-by: Chris Mason <chris.mason@oracle.com>

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index f3fd7e2..66b8341 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -511,6 +511,10 @@ static int btrfs_remount(struct super_block *sb, int *flags, char *data)
 	struct btrfs_root *root = btrfs_sb(sb);
 	int ret;
 
+	ret = btrfs_parse_options(root, data);
+	if (ret)
+		return -EINVAL;
+
 	if ((*flags & MS_RDONLY) == (sb->s_flags & MS_RDONLY))
 		return 0;
 


